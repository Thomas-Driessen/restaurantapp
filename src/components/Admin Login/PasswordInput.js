import { InputAdornment, withStyles } from '@material-ui/core';
import TextField from '@material-ui/core/TextField';
import { RemoveRedEye } from '@material-ui/icons';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

const styles = () => ({
  eye: {
    cursor: 'pointer',
  },
});

class PasswordInput extends Component {
  constructor() {
    super();

    this.state = {
      passwordIsMasked: true,
    };
  }

  togglePasswordMask = () => {
    this.setState(prevState => ({
      passwordIsMasked: !prevState.passwordIsMasked,
    }));
  };

  render() {
    const { classes } = this.props;
    const { passwordIsMasked } = this.state;

    return (
      <div className="inputField">
        <TextField
          type={passwordIsMasked ? 'password' : 'text'}
          id="outlined-search"
          variant="outlined"
          className='input'
          {...this.props}
          InputProps={{
            endAdornment: (
              <InputAdornment position="end">
                <RemoveRedEye
                  className={classes.eye}
                  onClick={this.togglePasswordMask}
                />
              </InputAdornment>
            ),
          }}
        />
      </div>
    );
  }
}

PasswordInput.propTypes = {
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.func.isRequired,
};

PasswordInput = withStyles(styles)(PasswordInput);

export default PasswordInput;