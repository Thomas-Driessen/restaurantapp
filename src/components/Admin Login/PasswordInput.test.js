import React from 'react';
import PasswordInput from './PasswordInput';
import renderer from 'react-test-renderer';

it('PasswordInput renders correctly', () => {
    const tree = renderer
        .create(<PasswordInput />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});