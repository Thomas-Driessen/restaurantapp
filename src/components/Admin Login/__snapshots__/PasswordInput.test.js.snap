// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput renders correctly 1`] = `
<div
  className="inputField"
>
  <div
    className="MuiFormControl-root MuiTextField-root input"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
        disabled={false}
        id="outlined-search"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root PasswordInput-eye-1"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </div>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-3"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
